src/ui/selectBox.js

```diff
@@ -0,0 +1,110 @@
+            optionItem: "<option value='%id' %selected>%name</option>\n"
+        },
+
+        includeNone: true,
+        noneLabel: "Select an Option",
+
+        model: {
+            label: "",
+            selectedItemId: false,
+            optionItems: {}
+        },
+
+        invokers: {
+            handleSelectChange: {
+                funcName: "youme.selectBox.handleSelectChange",
+                args: ["{that}", "{arguments}.0"]
+            },
+            renderMarkup: {
+                funcName: "youme.selectBox.render",
+                args: ["{that}", "{that}.options.markup.container"] // containerTemplate
+            },
+            renderOptions: {
+                funcName: "youme.selectBox.renderOptions",
+                args: ["{that}", "{that}.options.markup.optionItem" ] // itemTemplate
+            }
+        },
+        listeners: {
+            "onCreate.bind": {
+                this: "{that}.dom.selectBox",
+                method: "change",
+                args: ["{that}.handleSelectChange"]
+            }
+        },
+        modelListeners: {
+            selectBoxRefresh: {
+                excludeSource: "init",
+                path: ["selectedItemId", "optionItems"],
+                this: "{that}.dom.selectBox",
+                method: "html",
+                args: ["@expand:{that}.renderOptions()"]
+            },
+            label: {
+                excludeSource: "init",
+                this: "{that}.dom.label",
+                method: "html",
+                args: ["{that}.model.label"]
+            }
+        }
+    });
+
+    youme.selectBox.renderOptions = function (that, itemTemplate) {
+        var optionsContent = "";
+
+        if (that.options.includeNone) {
+            optionsContent += fluid.stringTemplate(itemTemplate, { id: "none", name: that.options.noneLabel });
+        }
+
+        fluid.each(that.model.optionItems, function (optionItem) {
+            var selected = optionItem.id === that.model.selectedItemId ? "selected" : "";
+            var mergedItemVariables = fluid.extend({}, optionItem, { selected: selected});
+            optionsContent += fluid.stringTemplate(itemTemplate, mergedItemVariables);
+        });
+        return optionsContent;
+    };
+
+    youme.selectBox.render = function (that, containerTemplate) {
+        var optionsContent = that.renderOptions();
+
+        var mergedContainerVariables = fluid.extend({}, that.model, { optionsContent: optionsContent});
+        var renderedContent = fluid.stringTemplate(containerTemplate, mergedContainerVariables);
+        return renderedContent;
+    };
+
+    youme.selectBox.handleSelectChange = function (that) {
+        var selectElement = that.locate("selectBox");
+        var selected = selectElement.val();
+        that.applier.change("selectedItemId", selected);
```

Comment:
Should be possible to do this using a new-style relay:
````
    modelRelay: {
        bindValue: {
            source: "dom.selectBox.value",
            target: "selectedItemId"
        }
    }
````

Reply:
Thanks, that's definitely a better pattern.